src/test/java/systemtests/EditEventCommandSystemTest.java

```diff
@@ -76,8 +76,8 @@ public void edit() {
         /* Case: redo editing the last calendar event in the list -> last calendar event edited again */
         command = RedoCommand.COMMAND_WORD;
         expectedResultMessage = RedoCommand.MESSAGE_SUCCESS;
-        model.updateCalendarEvent(
-                getModel().getFilteredCalendarEventList().get(INDEX_FIRST_ELEMENT.getZeroBased()), editedCalendarEvent);
+        model.updateCalendarEvent(getModel().getFilteredAndSortedCalendarEventList()
+                                            .get(INDEX_FIRST_ELEMENT.getZeroBased()), editedCalendarEvent);
         assertCommandSuccess(command, model, expectedResultMessage);
 
         /* Case: edit a calendar event with new values same as existing values -> edited */
```

Comment:
I think a test case we should add is
editing a calendar event's start date to after the end date -> rejected

Reply:
and maybe 1 test case for the natty date parsing

Reply:
I don't think natty needs a separate test case, cause whatever date strings we're feeding in will already be parsed by natty.

Reply:
agree that editing a calendar event's start date to after the end date

Reply:
Added a test case for the start/end order